### ISSUE TITLE
ringbuf put errors when running I2CBMP280.js and HeartRateDemo.js on A101

### ISSUE THREAD

**Issue by qiaojingx** [01org/zephyr.js#1582]
### Description

This is an regression issue and probably caused by #1542.` ringbuf put errors` prints when running [HeartRateDemo.js](https://github.com/01org/zephyr.js/blob/master/samples/HeartRateDemo.js) and [I2CBMP280.js](https://github.com/01org/zephyr.js/blob/master/samples/I2CBMP280.js) on A101.

### Test Code

[HeartRateDemo.js](https://github.com/01org/zephyr.js/blob/master/samples/HeartRateDemo.js);
[I2CBMP280.js](https://github.com/01org/zephyr.js/blob/master/samples/I2CBMP280.js)
 
### Steps to Reproduction

1. `make JS=samples/I2CBMP280.js`
2. `make dfu`

### Actual Result

![selection_029](https://user-images.githubusercontent.com/19813546/31429549-ef2b20d2-aea0-11e7-82c9-540a12ab0e0d.png)

### Expected Result

No error.
 
### Test Builds

Branch | Commit Id | Target Device | Test Date | Result
---|---|---|---|---
master | a36b2c9 - cd3f44e | Arduino101 | Oct 11, 2017 | Fail
master | 4b77505 | Arduino101 | Oct 11, 2017 | Pass

**Comment by brianjjones** [comment]
Looks like the setInterval isn't finishing before it gets called again.  I'll look to make sure there isn't a problem with callbacks.  It may also be that 10 is too low a timer number.

**Comment by brianjjones** [comment]
Looks like callbacks are fine, the demo timer is just going faster than the callbacks can be handled - hence the error message.  #1614 will fix this.

**Comment by qiaojingx** [comment]
Verified with commit cac4c3a + #1614. Error in I2CBMP280.js has been fixed. Could you change interval time in HeartRateDemo.js, too?

**Comment by grgustaf** [comment]
This seems to indicate we have a performance regression because we used to do this work in 10ms apparently and now it takes 200ms? Rather than mask the problem (:D) let's look into why. @brianjjones, could you try testing this just before and after #1542 to see if that's what caused the problem.

**Comment by grgustaf** [comment]
@brianjjones ping on this one, just note I don't think we're done here.

**Comment by jimmy-huang** [comment]
@cuiyanx is this still reproducible?

**Comment by cuiyanx** [comment]
@jimmy-huang 
Yes, this issue can be reproduced on both cases. And [/samples/Stepper.js](https://github.com/intel/zephyr.js/blob/master/samples/Stepper.js) case looks like this issue.
Log:
```
Stepper Motor sample...

554:(ERROR) 1 ringbuf put errors (last rval=-122)

554:(ERROR) 15 ringbuf put errors (last rval=-122)

554:(ERROR) 47 ringbuf put errors (last rval=-122)

554:(ERROR) 110 ringbuf put errors (last rval=-122)

```
